src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpOptions.cs

```diff
@@ -6,15 +6,24 @@ namespace Arcus.Messaging.Pumps.ServiceBus.Configuration
     /// <summary>
     /// The general options that configures a <see cref="AzureServiceBusMessagePump"/> implementation.
     /// </summary>
-    public class AzureServiceBusMessagePumpOptions
+    public class AzureServiceBusMessagePumpOptions : IAzureServiceBusQueueMessagePumpOptions, IAzureServiceBusTopicMessagePumpOptions
     {
         private int? _maxConcurrentCalls;
         private string _jobId = Guid.NewGuid().ToString();
         private TimeSpan _keyRotationTimeout = TimeSpan.FromSeconds(5);
         private int _maximumUnauthorizedExceptionsBeforeRestart = 5;
 
         /// <summary>
-        ///     Maximum concurrent calls to process messages
+        /// <para>Gets or sets the value indicating whether or not a new Azure Service Bus Topic subscription has to be created when the <see cref="AzureServiceBusMessagePump"/> starts.</para>
+        /// <para>The subscription will be deleted afterwards when the message pump stops if the options <see cref="ServiceBus.TopicSubscription.DeleteOnStop"/> is selected.</para>
+        /// </summary>
+        /// <remarks>
+        ///     Provides capability to create and delete these subscriptions. This requires 'Manage' permissions on the Azure Service Bus Topic or namespace.
+        /// </remarks>
+        public TopicSubscription TopicSubscription { get; set; } = TopicSubscription.CreateOnStart | TopicSubscription.DeleteOnStop;
```

Comment:
How is this different from the topic options?

Reply:
I changed this into a nullable enum so that it's more clear where it's used. The interfaces are only for consumer-friendly use to show what kind of options there's available for which entity type.